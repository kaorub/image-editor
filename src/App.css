.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
.App-body {
  height: 100%;
}
.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
/**
Flex column
 */
.App-column {
  display: flex;
  flex-direction: column;
  flex: 1;
}
/**
Flex direction
 */
.App-row {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;
}
/**
Vertical divider
 */
.App-divider {
  border-left: 1px solid grey;
  height: 500px;
}
/**
Image wrapper had double width
 */
.App-img-wrapper {
  flex: 2;
  position: relative;
}
