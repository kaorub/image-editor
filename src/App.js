import React from 'react';
import Button from './components/Button.js';
import './App.css';

function App() {
  return (
    <div className="App">
      <div className="App-body App-row">
          <div className="App-img-wrapper App-column">
            Here will be an image
          </div>
          <div className="App-divider"></div>
          <div className="App-actions-wrapper App-column">
              <Button name="new-test-block" value="new-test-block" label="new text block"/>
              <Button name="zoom-in" value="zoom-in" label="zoom in"/>
              <Button name="zoom-out" value="zoom-out" label="zoom out"/>
              <Button name="rotate" value="rotate" label="rotate"/>
          </div>
      </div>
    </div>
  );
}

export default App;
