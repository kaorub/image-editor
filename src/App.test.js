import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

test('snapshot renders', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
