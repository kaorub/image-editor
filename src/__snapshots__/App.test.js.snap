// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 1`] = `
<div
  className="App"
>
  <div
    className="App-body App-row"
  >
    <div
      className="App-img-wrapper App-column"
    >
      <img
        alt="to edit"
        className="Image"
        height={255}
        src="./images/sanfran.jpeg"
        width={255}
      />
    </div>
    <div
      className="App-divider"
    >
      â€‹
    </div>
    <div
      className="App-actions-wrapper App-column"
    >
      <div>
        <button
          className="Button"
          name="new-test-block"
          onClick={[Function]}
          value="new-test-block"
        >
          new text block
        </button>
      </div>
      <div>
        <button
          className="Button"
          name="zoom-in"
          onClick={[Function]}
          value="zoom-in"
        >
          zoom in
        </button>
      </div>
      <div>
        <button
          className="Button"
          name="zoom-out"
          onClick={[Function]}
          value="zoom-out"
        >
          zoom out
        </button>
      </div>
      <div>
        <button
          className="Button"
          name="rotate"
          onClick={[Function]}
          value="rotate"
        >
          rotate
        </button>
      </div>
    </div>
  </div>
</div>
`;
